File: DriftGame/libs/src/java-universal-tween-engine/tween-engine-api/src/aurelienribon/tweenengine/equations/Linear.java
package aurelienribon.tweenengine.equations;

import aurelienribon.tweenengine.TweenEquation;

/**
 * Easing equation based on Robert Penner's work: http://robertpenner.com/easing/
 */
public abstract class Linear extends TweenEquation {
    public static final Linear INOUT = new Linear() {
        @Override
        public float compute(float t) {
            return t;
        }

        @Override
        public String toString() {
            return "Linear.INOUT";
        }
    };
}

File: DriftGame/libs/src/java-universal-tween-engine/tween-engine-api/src/aurelienribon/tweenengine/equations/Linear.kt
package aurelienribon.tweenengine.equations

import aurelienribon.tweenengine.TweenEquation

/**
 * Easing equation based on [Robert Penner's work](http://robertpenner.com/easing/)
 */
abstract class Linear() : TweenEquation() {
    companion object {
        val INOUT = object : Linear() {
            override fun compute(t: Float) = t
        }
    }

    override fun toString() = "Linear.INOUT"
}